lib/api/v3/projects/schemas/project_schema_representer.rb

```diff
@@ -107,8 +109,75 @@ class ProjectSchemaRepresenter < ::API::Decorators::SchemaRepresenter
           schema :updated_at,
                  type: "DateTime"
 
-          def self.represented_class
-            ::Project
+          # Cannot be cached here due to custom field visibility checks on a user level.
+          # However caching is still applied further down in the `section_representation` method.
+          property :attribute_groups,
+                   name_source: ->(*) { I18n.t("activerecord.attributes.project.attribute_groups") },
+                   type: "[]String",
+                   as: "_attributeGroups",
+                   exec_context: :decorator,
+                   uncacheable: true
+
+          class << self
+            def represented_class
+              ::Project
+            end
+
+            def attribute_group(property)
+              lambda do
+                key = property.to_s
+                attribute_group_map key
+              end
+            end
+          end
+
+          def attribute_groups
+            project_custom_field_sections.map do |section|
+              section_representation(section)
+            end
+          end
+
+          ##
+          # Return a map of attribute => group name
+          def attribute_group_map(key)
+            return nil if project_custom_field_sections.empty?
+
+            @attribute_group_map ||= project_custom_field_sections.each_with_object({}) do |section, hash|
+              section.custom_fields.each do |cf|
+                hash[cf.attribute_name(:camel_case)] = section.name
+              end
+            end
+
+            @attribute_group_map[key]
+          end
+
+          private
+
+          def project_custom_field_sections
+            @project_custom_field_sections ||= ProjectCustomFieldSection
+                                                .joins(:custom_fields)
+                                                .includes(:custom_fields)
+                                                .merge(ProjectCustomField.visible(current_user))
+                                                .group(:id, "custom_fields.id")
+                                                .order(:position, :position_in_custom_field_section)
+          end
+
+          def section_representation(section)
+            OpenProject::Cache.fetch(*section_cache_key(section)) do
+              ::JSON::parse(::API::V3::Projects::Schemas::ProjectCustomFieldSectionRepresenter
+                              .new(section, current_user:, embed_links: true)
+                              .to_json)
```

Comment:
Why convert to json than parse?
```suggestion
              ::API::V3::Projects::Schemas::ProjectCustomFieldSectionRepresenter
                              .new(section, current_user:, embed_links: true)
```

Reply:
That's a good question, that pattern has been established [here](https://github.com/opf/openproject/blob/6d32400044979cd28a9a0d87dff9d436a0ecc828/lib/api/v3/work_packages/schema/work_package_schema_representer.rb#L367-L384).
At least in this instance I can safely remove it.